sys/kern/kern_lock.c

```diff
@@ -1462,9 +1475,11 @@ __lockmgr_args(struct lock *lk, u_int flags, struct lock_object *ilk,
 				continue;
 			}
 
-			v = x & (LK_ALL_WAITERS | LK_EXCLUSIVE_SPINNERS);
-			if ((x & ~v) == LK_UNLOCKED) {
-				v = (x & ~LK_EXCLUSIVE_SPINNERS);
+			v = cheri_get_low_ptr_bits(x, LK_ALL_WAITERS |
+			    LK_EXCLUSIVE_SPINNERS);
+			if (cheri_clear_low_ptr_bits(x, v) == LK_UNLOCKED) {
```

Comment:
Ah I guess this one does need the macros.

Reply:
I don't see why?